examples/viewer.py

```diff
@@ -333,26 +369,56 @@ def reconfigure_sim(self) -> None:
         self.agent_id: int = self.sim_settings["default_agent"]
         self.cfg.agents[self.agent_id] = self.default_agent_config()
 
+        if self.enable_batch_renderer:
+            self.cfg.sim_cfg.create_renderer = False
+            self.cfg.sim_cfg.enable_gfx_replay_save = True
+
         if self.sim_settings["stage_requires_lighting"]:
             logger.info("Setting synthetic lighting override for stage.")
             self.cfg.sim_cfg.override_scene_light_defaults = True
             self.cfg.sim_cfg.scene_light_setup = habitat_sim.gfx.DEFAULT_LIGHTING_KEY
 
         if self.sim is None:
-            self.sim = habitat_sim.Simulator(self.cfg)
-
+            self.sim = []
+            for _i in range(self.num_env):
+                self.sim.append(habitat_sim.Simulator(self.cfg))
         else:  # edge case
-            if self.sim.config.sim_cfg.scene_id == self.cfg.sim_cfg.scene_id:
-                # we need to force a reset, so change the internal config scene name
-                self.sim.config.sim_cfg.scene_id = "NONE"
-            self.sim.reconfigure(self.cfg)
+            for i in range(self.num_env):
+                if self.sim[i].config.sim_cfg.scene_id == self.cfg.sim_cfg.scene_id:
+                    # we need to force a reset, so change the internal config scene name
+                    self.sim[i].config.sim_cfg.scene_id = "NONE"
+                self.sim[i].reconfigure(self.cfg)
+
         # post reconfigure
-        self.active_scene_graph = self.sim.get_active_scene_graph()
-        self.default_agent = self.sim.get_agent(self.agent_id)
-        self.agent_body_node = self.default_agent.scene_node
-        self.render_camera = self.agent_body_node.node_sensor_suite.get("color_sensor")
+        self.default_agent = []
+        self.render_camera = []
+        for i in range(self.num_env):
+            self.default_agent.append(self.sim[i].get_agent(self.agent_id))
+            self.render_camera.append(
+                self.default_agent[i].scene_node.node_sensor_suite.get("color_sensor")
+            )
+
         # set sim_settings scene name as actual loaded scene
-        self.sim_settings["scene"] = self.sim.curr_scene_name
+        self.sim_settings["scene"] = self.sim[0].curr_scene_name
```

Comment:
In general I'd like to see more code in viewer.py like this that can just assume one sim. Maybe a helper, `self.get_sim()` or `self.default_sim`. Or maybe just keep `self.sim` and add `self.tiled_sims` for the niche tiled use-case.

Again, my high-level point is that 95% of viewer.py users don't need or care about multiple envs, and I'd like to keep most of the code in this file simpler.